apollo-router/src/axum_factory/compression/mod.rs

```diff
@@ -79,7 +79,9 @@ where {
                         }
                     }
                     Ok(data) => {
-                        let mut buf = BytesMut::zeroed(data.len());
+                        // the buffer needs at least 10 bytes for a gzip header if we use gzip, then more
+                        // room to store the data itself
+                        let mut buf = BytesMut::zeroed(10 + data.len());
```

Comment:
10 should be a const GZIP_HEADER_LEN.

Reply:
done in [374a868](https://github.com/apollographql/router/pull/3961/commits/374a8688cb6b34b927af839388e3dafd8335e72d)